examples/2d/mesh2d_material.rs

```diff
@@ -0,0 +1,119 @@
+use bevy::{
+    ecs::system::{lifetimeless::SRes, SystemParamItem},
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        render_asset::{PrepareAssetError, RenderAsset},
+        render_resource::{
+            std140::{AsStd140, Std140},
+            *,
+        },
+        renderer::RenderDevice,
+    },
+    sprite::Material2d,
+    sprite::{Material2dPipeline, Material2dPlugin, MaterialMesh2dBundle, Mesh2dHandle},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_plugin(Material2dPlugin::<Custom2dMaterial>::default())
+        .add_startup_system(setup)
+        .run();
+}
+
+/// set up a simple 3D scene
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<Custom2dMaterial>>,
+) {
+    let size = Vec2::new(300.0, 300.0);
+
+    let material = materials.add(Custom2dMaterial {
+        color: Vec4::new(0.65, 0.0, 0.5, 1.0),
+    });
+
+    // quad
+    commands.spawn().insert_bundle(MaterialMesh2dBundle {
+        mesh: Mesh2dHandle(meshes.add(Mesh::from(shape::Quad::new(size)))),
+        material,
+        ..Default::default()
+    });
+
+    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
+}
+
+#[derive(Debug, Clone, TypeUuid, AsStd140)]
+#[uuid = "4ee9c363-1124-4113-890e-199d81b00281"]
```

Comment:
Use a different UUID than the 3d example please, it's okay here since they are never in the same program but the should be different nonetheless